### PR TITLE
Add HTML pretty printing and tests

### LINKED ISSUE #34: Pretty printing HTML
Currently the produced HTML doesn't look good: it inserts a newline before each attribute. It would be nice to have a pretty printing option.

I am willing to work on this in a PR. My current plan is to change the `to_string` signature to:
```ocaml
val to_string : xml:bool -> indent_with:[< `Tabs | `Spaces of int ] -> node -> string
```
The algorithm would be roughly this:
- If current tag has no children: open and close the tag (appropriately), without a newline.
- If current tag's children has at least one `Txt` node: do not indent the children.
- Otherwise, indent the children.

The second rule is like that for cases like: `<p><b>Hello,</b> World</p>` should be pretty printed to the same thing. However, `<p><b>Hello, World</b></p>` should be pretty printed to:
```
<p>
  <b>Hello, World</b>
</p>
```
Basically, if all the children are tags or comments, we can safely indent them.

Let me know your inputs.

### PR DESCRIPTION
**Pull request by RezwanArefin01**
Closes #34. Changes:
- `Pure_html.to_string` now pretty prints using two spaces for indentation.
- The tests are updated to use expect tests, as the previous setup was hard to work with.
  - `dream_html_tests.ml` actually had wrong test cases (eg. wrong outputs accepted as expected outputs)! Fixed those as well. Now it is much easier to see what is going on in the tests due to using expect tests!
- Added some corner cases to test the pretty printing.

For review:
- It might be better to read the new code instead of the diff. The diff seems a bit messy.

### CODE REVIEW

File: pure-html/pure_html.ml [hunk 1]
```diff
@@ -51,41 +59,101 @@ let xml_mode xml name =
...
+    | None -> ""
+    | Some _ -> "\n"
+end
+
+(* Loosely based on https://www.w3.org/TR/DOM-Parsing/. Pretty prints using two
+  spaces for indentation. On high level, the algorithm indents the children of a
+  tag if they do not contain a txt node. No newline is inserted if there are no
+  children nodes. *)
+let rec write_tag ~indent_level ~xml p node =
+  let should_indent_next children =
```
Comment (yawaramin) [inline pure-html/pure_html.ml:100]: This function doesn't need to be nested; we can lift it up.

File: test/dune [hunk 2]
```diff
@@ -1,23 +1,6 @@
...
-  (diff expected.txt got.txt)))
-
-(rule
- (with-stdout-to
-  got.txt
-  (run ./dream_html_test.exe)))
+ (inline_tests)
+ (libraries dream_html pure_html fmt)
+ (preprocess
+  (pps ppx_expect)))
```
Comment (yawaramin) [inline test/dune:6]: Can you also add this as a test dependency of the `pure-html` package in the `dune-project` file? As that will be needed to install correctly for CI. Example of how to do that: https://github.com/yawaramin/ocaml-decimal/blob/ddef44e4c625ee507aab7bddb6350bd092cc475e/dune-project#L29

Remember to run `dune build` once after updating the `dune-project` file so that it can regenerate the opam files. Thanks!

**Review by yawaramin — COMMENTED** [review]
Nice work! I only have a few small comments. We should be able to merge soon.

File: pure-html/pure_html.ml [hunk 3]
```diff
@@ -51,41 +59,101 @@ let xml_mode xml name =
   | true, _ | _, ("math" | "svg") -> true
   | _ -> false
 
-(* Loosely based on https://www.w3.org/TR/DOM-Parsing/ *)
-let rec write_tag ~xml p = function
+module Indent_level = struct
+  type t = int option
+
+  let next (t : t) =
```
[the remark this answers is not in the document]
Comment (RezwanArefin01) [reply pure-html/pure_html.ml]: Updated.

File: pure-html/pure_html.ml (hunk 1 above)
Comment (RezwanArefin01) [reply pure-html/pure_html.ml]: Updated.

File: test/dune (hunk 2 above)
Comment (RezwanArefin01) [reply test/dune]: Also `ppx_expect`. Although removing `fmt` from the `(libraries)` stanza compiles, because it already indirectly depends on it: `dream_html -> dream -> fmt`.

Changes:
- Separated dream-html and pure-html tests into two libraries (otherwise both would try to run for pure-html runtest).
- Bumped dune language version to 2.8, which is the first version that allows `(library (package ...))`.

**Review by yawaramin — APPROVED** [review]
Great, thanks!
